[serviceNow] Add details on filtering API (#1944)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/incident-management/servicenow.md

The resources that can be ingested from ServiceNow into Port are listed below. 

- `Group` - (`<your-servicenow-url>/api/now/table/sys_user_group`)
- `Service Catalog` - (`<your-servicenow-url>/api/now/table/sc_catalog`)
- `Incident` - (`<your-servicenow-url>/api/now/table/incident`)

:::tip Ingesting extra resources

> While our documentation lists only three supported resources, Port's ServiceNow integration uses the [ServiceNow Table API](https://developer.servicenow.com/dev.do#!/reference/api/xanadu/rest/c_TableAPI#table-GET) to ingest entities. This means you can ingest a lot more resources from your ServiceNow instance as long as the underlying resource can be found in the Table API. All you need is to specify the `table name` as a new `kind` in the [Data sources configuration page](/build-your-software-catalog/sync-data-to-catalog/#customize-your-integrations), and the records from the table will be ingested to Port.

Reviewer: ```suggestion
While the section above only lists three supported resources, Port's ServiceNow integration uses the [ServiceNow Table API](https://developer.servicenow.com/dev.do#!/reference/api/xanadu/rest/c_TableAPI#table-GET) to ingest entities.  
This means you can ingest a lot more resources from your ServiceNow instance as long as the underlying resource can be found in the Table API. All you need is to specify the `table name` as a new `kind` in the [Data sources configuration page](/build-your-software-catalog/sync-data-to-catalog/#customize-your-integrations), and the records from the table will be ingested to Port.
```

</details>

### Filtering ServiceNow resources
Port's ServiceNow integration provides an option to filter the data that is retrieved from the ServiceNow Table API using the following attributes:

1. `sysparmDisplayValue`: Determines the type of data returned, either the actual values from the database or the display values of the fields. The default is `true`
2. `sysparmFields`: Comma-separated list of fields to return in the response
3. `sysparmExcludeReferenceLink`: Flag that indicates whether to exclude Table API links for reference fields. The default is `false`
4. `sysparmQuery`: Encoded query used to filter the result set. The syntax is `<col_name><operator><value>`:
    1. `<col_name>`: Name of the table column to filter against
    2. `<operator>`: =, !=, ^, ^OR, LIKE, STARTSWITH, ENDSWITH, `ORDERBY<col_name>`, `ORDERBYDESC<col_name>`
    3. `<value>`: Value to match against

    Queries can be chained using ^ or ^OR for AND/OR logic. An example query could be this: `active=true^nameLIKEdev^urgency=3` which returns all active incidents with an urgency level of 3 and have a name like `dev` 

> All these four filtering attributes can be enabled using the path: `selector.apiQueryParams`.

Reviewer: this is not really clear, how does the user enable them? perhaps add a small example

</details>

### Filtering ServiceNow resources
Port's ServiceNow integration provides an option to filter the data that is retrieved from the ServiceNow Table API using the following attributes:

1. `sysparmDisplayValue`: Determines the type of data returned, either the actual values from the database or the display values of the fields. The default is `true`
2. `sysparmFields`: Comma-separated list of fields to return in the response
3. `sysparmExcludeReferenceLink`: Flag that indicates whether to exclude Table API links for reference fields. The default is `false`
4. `sysparmQuery`: Encoded query used to filter the result set. The syntax is `<col_name><operator><value>`:
    1. `<col_name>`: Name of the table column to filter against
    2. `<operator>`: =, !=, ^, ^OR, LIKE, STARTSWITH, ENDSWITH, `ORDERBY<col_name>`, `ORDERBYDESC<col_name>`
    3. `<value>`: Value to match against

    Queries can be chained using ^ or ^OR for AND/OR logic. An example query could be this: `active=true^nameLIKEdev^urgency=3` which returns all active incidents with an urgency level of 3 and have a name like `dev` 

> All these four filtering attributes can be enabled using the `selector.apiQueryParams` path as shown below:

Reviewer: ```suggestion
The filtering attributes described above can be enabled using the `selector.apiQueryParams` path, for example:
```
